store/orm/orm.go

```diff
@@ -5,86 +5,68 @@ import (
 	"errors"
 	"fmt"
 	"math/big"
-	"reflect"
-	"sort"
 	"time"
 
-	"github.com/asdine/storm"
-	"github.com/asdine/storm/index"
-	"github.com/asdine/storm/q"
 	"github.com/ethereum/go-ethereum/common"
 	"github.com/ethereum/go-ethereum/core/types"
-	"github.com/smartcontractkit/chainlink/logger"
+	"github.com/jinzhu/gorm"
+	_ "github.com/jinzhu/gorm/dialects/sqlite" // follows the gorm convention
 	"github.com/smartcontractkit/chainlink/store/models"
 	"github.com/smartcontractkit/chainlink/utils"
-	bolt "go.etcd.io/bbolt"
 	"go.uber.org/multierr"
 )
 
 var (
-	// ErrorInvalidCallbackSignature is returned in AllInBatches if the incorrect function signature is passed.
-	ErrorInvalidCallbackSignature = errors.New("AllInBatches callback has incorrect function signature, must return bool")
-	// ErrorInvalidCallbackModel is returned in AllInBatches if the model and bucket do not match types.
-	ErrorInvalidCallbackModel = errors.New("AllInBatches callback has incorrect model, must match bucket")
 	// ErrorNotFound is returned when finding a single value fails.
-	ErrorNotFound = storm.ErrNotFound
+	ErrorNotFound = gorm.ErrRecordNotFound
 )
 
 // ORM contains the database object used by Chainlink.
 type ORM struct {
-	*storm.DB
+	DB *gorm.DB
 }
 
 // NewORM initializes a new database file at the configured path.
-func NewORM(path string, duration time.Duration) (*ORM, error) {
-	db, err := initializeDatabase(path, duration)
+func NewORM(path string) (*ORM, error) {
+	db, err := initializeDatabase(path)
 	if err != nil {
 		return nil, fmt.Errorf("unable to init DB: %+v", err)
 	}
-	return &ORM{db}, nil
+	return &ORM{DB: db}, nil
 }
 
-func initializeDatabase(path string, duration time.Duration) (*storm.DB, error) {
-	options := storm.BoltOptions(0600, &bolt.Options{Timeout: duration})
-	db, err := storm.Open(path, options)
+func initializeDatabase(path string) (*gorm.DB, error) {
+	db, err := gorm.Open("sqlite3", path)
 	if err != nil {
-		return nil, fmt.Errorf("unable to open stormDB: %+v", err)
+		return nil, fmt.Errorf("unable to open gorm DB: %+v", err)
 	}
 	return db, nil
 }
 
-// GetBolt returns BoltDB from the ORM
-func (orm *ORM) GetBolt() *bolt.DB {
-	return orm.DB.Bolt
+func multifyWithoutRecordNotFound(db *gorm.DB) error {
```

Comment:
Stumbled over `multify`, googled for it a bit before realizing it was multierr related. Maybe something like `ignoreNotFoundErrors`?